rabbitmq-server-4.0.yaml

```diff
@@ -150,14 +153,54 @@ subpackages:
 
 update:
   enabled: true
+  ignore-regex-patterns:
+    - .*alpha.*
+    - .*beta.*
+    - .*rc.*
   github:
     identifier: rabbitmq/rabbitmq-server
     strip-prefix: v
     tag-filter-prefix: v4.0.
 
 test:
+  environment:
+    contents:
+      packages:
+        - curl
+        - shadow
+        - sudo-rs
   pipeline:
-    # AUTOGENERATED
-    - runs: |
-        rabbitmq-plugins version
-        rabbitmq-plugins --version
+    - uses: test/daemon-check-output
+      with:
+        setup: |
+          groupadd -r rabbitmq
+          useradd -r -g rabbitmq -d /var/lib/rabbitmq -s /sbin/nologin rabbitmq
+
+          mkdir -p /var/lib/rabbitmq /var/log/rabbitmq /etc/rabbitmq
+
+          # Set Erlang cookie
+          echo "supersecreterlangcookie" > /var/lib/rabbitmq/.erlang.cookie
+          chmod 600 /var/lib/rabbitmq/.erlang.cookie
+          chown -R rabbitmq:rabbitmq /var/lib/rabbitmq /var/log/rabbitmq /etc/rabbitmq
+          export HOME=/var/lib/rabbitmq
+          RABBITMQ_LOGS=/var/log/rabbitmq/rabbit.log
+          RABBITMQ_SASL_LOGS=/var/log/rabbitmq/rabbit-sasl.log
+          export RABBITMQ_ENABLED_PLUGINS_FILE=/etc/rabbitmq/enabled_plugins
+          mkdir -p /etc/rabbitmq
+          echo '[rabbitmq_management].' > $RABBITMQ_ENABLED_PLUGINS_FILE
+        start: rabbitmq-server
+        timeout: 40
+        expected_output: |
+          RabbitMQ ${{package.version}}
+          Starting broker...
+        post: |
+          # wait for the management plugin to start
+          sleep 3
```

Comment:
nit: If we use wait-for-it here to wait for port to come up then that'll be likely less flaky and easy maintenance. 

We can do something like
```
wait-for-it -t 30 localhost:15672 -- echo "management UI is up"
``

We'll have to add wait-for-it inside test environment for this.

Reply:
This sounds better will add it, thanks for suggesting this